docs/build.sh

```diff
@@ -16,14 +16,21 @@ if [[ ! $(command -v jq) ]]; then
     exit 1
 fi
 
+MAIN_BRANCH="refs/heads/main"
 RELEASE_TAG_REGEX="^refs/tags/v.*"
 
 if [[ "${AZP_BRANCH}" =~ ${RELEASE_TAG_REGEX} ]]; then
     DOCS_TAG="${AZP_BRANCH/refs\/tags\//}"
     export DOCS_TAG
+    # no need to build rst explicitly, just html
+    HTML_ONLY=true
 else
     BUILD_SHA=$(git rev-parse HEAD)
     export BUILD_SHA
+    if [[ "${AZP_BRANCH}" == "${MAIN_BRANCH}" ]]; then
+        # no need to build html, just rst
+        RST_ONLY=true
+    fi
```

Comment:
not sure why but building both rst and html targets bazel seems to rerun the protodoc aspect build

as html deps on the rst target (and building html there is an rst.tar in bazel's cache) i had hoped it would be v quick to build both

im going to try reordering the build commands to see if it makes any difference, if not i think we can just set it to build html only in all normal runs, and the rst only target in main runs so the latest docs can be published from it